### PR TITLE
Add Plugwise component

### PR DESCRIPTION
**Pull request by CoMPaTech**
**Description:**

Adding Plugwise thermostats to be controlled and monitored through an integration. Including the other `climate` products of this vendor is anticipated by the community, but let's start with this (current `custom_component`) and use our community forces to expand before the PRs get too big.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#25533

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

--

Waiting on original author of `haanna.py` dependency to update PyPi module before removing [WIP].

--

As this is my first (set of PRs) please correct me where necessary - I'm on discord as well.

<a href="https://gitpod.io/#https://github.com/home-assistant/home-assistant.io/pull/9992"></a>

### CODE REVIEW

**Comment by CoMPaTech** [comment]
At least (or last), home-assistant/home-assistant#25533 passed it's test-cycle

File: source/_components/plugwise.markdown [hunk 1]
```diff
@@ -0,0 +1,149 @@
+---
+title: "Plugwise"
+description: "Plugwise Climate integration."
+logo: plugwise.png
+ha_category: Climate
+ha_iot_class: "Local Polling"
```
Comment (klaasnicolaas) [inline source/_components/plugwise.markdown]:
```suggestion
ha_iot_class: Local Polling
```

File: source/_components/plugwise.markdown [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+redirect_from:
+  - /components/climate.tado/
```
Comment (klaasnicolaas) [inline source/_components/plugwise.markdown]: Why a redirect from tado?

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_components/plugwise.markdown (hunk 2 above)
Comment (CoMPaTech) [reply source/_components/plugwise.markdown]: Should have removed that (was using other climates as reference), my sincere apologies!

**Comment by CoMPaTech** [comment]
@klaasnicolaas applied the changes you indicated, thanks for the review! Is including the image ok? (Didn't spot that in the docs on the developer pages, but it seemed logical enough to include it :))

File: source/_components/plugwise.markdown [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+```yaml
+# Minimal configuration.yaml entry
+climate:
+  - platform: plugwise
+    password: your_short_id # required, the ID on the smile (some string of 6 characters)
+    host: local_ip_address  # required, the IP-address of your smile
+```
+
+```yaml
+# Example configuration.yaml entry to show only devices at your vacation and primary homes
```
Comment (klaasnicolaas) [inline source/_components/plugwise.markdown]: Maybe move this example below the configuration variables with a `Full example` title?

Comment (CoMPaTech) [reply source/_components/plugwise.markdown]: I've commited changes (and did the same with the service calls)

**Comment by CoMPaTech** [comment]
Changed title to match PR for integration

**Comment by CoMPaTech** [comment]
@klaasnicolaas '1 change requested' is probably something that doesn't auto-remove when things are `resolved`, right?

**Comment by klaasnicolaas** [comment]
That's because it wasn't a change request but a simple added note

File: source/_components/plugwise.markdown (hunk 3 above)
Comment (klaasnicolaas) [reply source/_components/plugwise.markdown]: I see the full example still above the `configuration variables`

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_components/plugwise.markdown (hunk 3 above)
Comment (CoMPaTech) [reply source/_components/plugwise.markdown]: Marking unresolved, my apologies.

Moved it accordingly now.

**Review by klaasnicolaas — DISMISSED** [review]
LGTM 😉

**Comment by CoMPaTech** [comment]
Tnx!

File: source/_components/plugwise.markdown [hunk 4]
```diff
@@ -0,0 +1,156 @@
+---
+title: "Plugwise"
+description: "Plugwise Climate integration."
+logo: plugwise.png
+ha_category: Climate
+ha_iot_class: Local Polling
+ha_release: 0.97
```
Comment (MartinHjelmare) [inline source/_components/plugwise.markdown]: 0.98

File: source/_components/plugwise.markdown [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+---
+
+This enables [Plugwise](https://plugwise.com) [Anna](https://www.plugwise.com/en_US/products/anna) thermostats to be integrated. This integration talks locally to your **Smile** interface, you will need it's secret key and IP address.
```
Comment (MartinHjelmare) [inline source/_components/plugwise.markdown]: you will need its secret key

File: source/_components/plugwise.markdown [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+  description: The IP address of your Smile. 
+  required: true
+  type: string
+name:
+  description: The name of your thermostat, i.e. "Anna"
+  required: false
+  type: string
+username:
+  description: Should you ever need to change this, you can
+  required: false
```
Comment (MartinHjelmare) [inline source/_components/plugwise.markdown:46]: Add the default value using the `default` key.

File: source/_components/plugwise.markdown [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+  type: string
+port:
+  description: When having a custom setup, you can change the port number
+  required: false
+  type: integer
+min_temp:
+  description: If you want to adjust the lower boundary, the integration will no allow temperatures below the set value.
```
Comment (MartinHjelmare) [inline source/_components/plugwise.markdown]: the integration will not allow temperatures

File: source/_components/plugwise.markdown [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+```
+
+### Service
+
+#### Set HVAC mode (schedule)
+
+Service: `climate.set_hvac_mode`
+
+Available options include `auto` or `off`. Meaning `auto` is a schedule is active and `off` there is no schedule active (active preset or manually set temperature will continue).
+The last schedule that was active is determined the same way long-tapping the top of Anna works.
```
Comment (MartinHjelmare) [inline source/_components/plugwise.markdown:86]: The grammar in this section isn't 💯. There are some connecting words missing.

Perhaps:

"The meaning of `auto` is that a schedule is active and the meaning of `off` is that there is no schedule active, i.e. that an active preset or manually set temperature will continue."

**Review by MartinHjelmare — CHANGES_REQUESTED** [review]

File: source/_components/plugwise.markdown (hunk 6 above)
Comment (CoMPaTech) [reply source/_components/plugwise.markdown:46]: Added all the defaults accordingly

File: source/_components/plugwise.markdown (hunk 8 above)
Comment (CoMPaTech) [reply source/_components/plugwise.markdown:86]: Used your suggestion and tried on making it more clear

File: source/_components/plugwise.markdown [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+    max_temp: 30 	    # optional, when you want to change the maximum temperature
+```
+
+### Service
+
+#### Set HVAC mode (schedule)
+
+Service: `climate.set_hvac_mode`
+
+Available options include `auto` or `off`. The meaning of `auto` is that a schedule is active and the thermostat will change presets accordingly. The meaning of `off` there is no schedule active, i.e. the active preset or manually set temperature be used to control the climate of your house/rooms.
```
Comment (MartinHjelmare) [inline source/_components/plugwise.markdown]: The meaning of `off` is that there is no schedule active, i.e. the active preset or manually set temperature is to be used to control the climate of your house or rooms.

File: source/_components/plugwise.markdown [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+    max_temp: 30 	    # optional, when you want to change the maximum temperature
+```
+
+### Service
+
+#### Set HVAC mode (schedule)
+
+Service: `climate.set_hvac_mode`
+
+Available options include `auto` or `off`. The meaning of `auto` is that a schedule is active and the thermostat will change presets accordingly. The meaning of off is that there is no schedule active, i.e. the active preset or manually set temperature is to be used to control the climate of your house or rooms.
```
Comment (MartinHjelmare) [inline source/_components/plugwise.markdown]: Wrap `off` in backtics the second occurrence too as done for `auto`.

**Review by MartinHjelmare — DISMISSED** [review]

File: source/_components/plugwise.markdown [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+description: "Plugwise Climate integration."
+logo: plugwise.png
+ha_category: Climate
+ha_iot_class: Local Polling
+ha_release: 0.98
+---
+
+This enables [Plugwise](https://plugwise.com) [Anna](https://www.plugwise.com/en_US/products/anna) thermostats to be integrated. This integration talks locally to your **Smile** interface, you will need its password and IP address.
+
+The password can be found on the bottom of your Smile, it should consist of 6 characters. To find your IP address use the Plugwise App:
```
Comment (frenck) [inline source/_components/plugwise.markdown:12]: Ensure there is a new line before starting a list.
```suggestion
The password can be found on the bottom of your Smile, it should consist of 6 characters. To find your IP address use the Plugwise App: 

```

File: source/_components/plugwise.markdown [hunk 12]
```diff
@@ -0,0 +1,164 @@
+---
+title: "Plugwise"
+description: "Plugwise Climate integration."
+logo: plugwise.png
+ha_category: Climate
+ha_iot_class: Local Polling
+ha_release: 0.98
+---
+
+This enables [Plugwise](https://plugwise.com) [Anna](https://www.plugwise.com/en_US/products/anna) thermostats to be integrated. This integration talks locally to your **Smile** interface, you will need its password and IP address.
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
This enables [Plugwise](https://plugwise.com) [Anna](https://www.plugwise.com/en_US/products/anna) thermostats to be integrated. This integration talks locally to your **Smile** interface, and you will need its password and IP address.
```

File: source/_components/plugwise.markdown [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+
+The password can be found on the bottom of your Smile, it should consist of 6 characters. To find your IP address use the Plugwise App: 
+ - Open the Plugwise App and choose the 'Settings'-icon (&#9776;) and choose 'HTML-interface'. 
+ - Go to the (lower) 'Settings'-icon (&#9776;) and choose 'Preferences'. 
+ - Choose 'System' then 'Networking' and your IP address will be shown.
+
+## Configuration
+
+You have to add the following to your configuration.yaml file.
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
You have to add the following to your configuration.yaml file:
```

File: source/_components/plugwise.markdown [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+
+```yaml
+# Minimal configuration.yaml entry
+climate:
+  - platform: plugwise
+    password: your_short_id # required, the ID on the smile (some string of 6 characters)
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
    password: YOUR_SHORT_ID
```

Placeholders need to be in capital letters, descriptions and instructions go into the configuration section, so please remove those comments.

File: source/_components/plugwise.markdown [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+    host: local_ip_address  # required, the IP-address of your smile
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
    host: YOUR_SMILE_LOCAL_IP
```

Placeholders need to be in capital letters, descriptions and instructions go into the configuration section, so please remove those comments.

File: source/_components/plugwise.markdown [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+```
+
+{% configuration %}
+password:
+  description: Your Smile ID (located on the bottom of the Smile, not the Anna)
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
  description: Your Smile ID (located on the bottom of the Smile, not the Anna).
```

File: source/_components/plugwise.markdown [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+  required: true
+  type: string
+host:
+  description: The IP address of your Smile. 
+  required: true
+  type: string
+name:
+  description: The name of your thermostat, i.e. "Anna"
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
  description: The name of your thermostat, i.e., "Anna".
```

File: source/_components/plugwise.markdown [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+  required: false
+  type: string
+  default: "Plugwise Thermostat"
+username:
+  description: Should you ever need to change this, you can
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
  description: Should you ever need to change this, you can.
```

File: source/_components/plugwise.markdown [hunk 19]
```diff
@@ -0,0 +1,164 @@
...
+  required: false
+  type: string
+  default: smile
+port:
+  description: When having a custom setup, you can change the port number
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
  description: When having a custom setup, you can change the port number.
```

File: source/_components/plugwise.markdown [hunk 20]
```diff
@@ -0,0 +1,164 @@
...
+  required: false
+  type: integer
+  default: 30
+{% endconfiguration %}
+
+### Full configuration example
+
+```yaml
+# Example configuration.yaml entry to show only devices at your vacation and primary homes
+climate:
```
Comment (frenck) [inline source/_components/plugwise.markdown:69]: Remove all comments, they should be part of the config section.
Make sure place holders are in all caps.

File: source/_components/plugwise.markdown [hunk 21]
```diff
@@ -0,0 +1,164 @@
...
+    max_temp: 30 	    # optional, when you want to change the maximum temperature
+```
+
+### Service
+
+#### Set HVAC mode (schedule)
+
+Service: `climate.set_hvac_mode`
+
+Available options include `auto` or `off`. The meaning of `auto` is that a schedule is active and the thermostat will change presets accordingly. The meaning of `off` is that there is no schedule active, i.e. the active preset or manually set temperature is to be used to control the climate of your house or rooms.
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
Available options include `auto` or `off`. The meaning of `auto` is that a schedule is active and the thermostat will change presets accordingly. The meaning of `off` is that there is no schedule active, i.e., the active preset or manually set temperature is to be used to control the climate of your house or rooms.
```

File: source/_components/plugwise.markdown [hunk 22]
```diff
@@ -0,0 +1,164 @@
...
+      - service: climate.set_preset_mode
+        data:
+          preset_mode: asleep
+```
+
+### Troubleshooting
+
+Example of a working configuration excerpt (with debugging enabled):
+
+```source
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
```txt
```

File: source/_components/plugwise.markdown [hunk 23]
```diff
@@ -0,0 +1,164 @@
...
+[homeassistant.components.climate] Setting up climate.plugwise
+[custom_components.plugwise.climate] Init called
+[custom_components.plugwise.climate] Initializing API
+[custom_components.plugwise.climate] platform ready
+[custom_components.plugwise.climate] Update called
+```
+
+Example of something going wrong (IP address not set) excerpt is shown below. Correct your configuration and try again. If the errors persist, please share a larger excerpt of your logfile
+
+```source
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
```txt
```

File: source/_components/plugwise.markdown [hunk 24]
```diff
@@ -0,0 +1,164 @@
...
+[homeassistant.loader] You are using a custom integration for plugwise which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.
+[custom_components.plugwise.climate] Plugwise: custom component loading (Anna PlugWise climate)
+[homeassistant.components.climate] Setting up climate.plugwise
+[custom_components.plugwise.climate] Init called
+[custom_components.plugwise.climate] Initializing API
+[custom_components.plugwise.climate] platform ready
+[custom_components.plugwise.climate] Update called
+```
+
+Example of something going wrong (IP address not set) excerpt is shown below. Correct your configuration and try again. If the errors persist, please share a larger excerpt of your logfile
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
Example of something going wrong (IP address not set) excerpt is shown below. Correct your configuration and try again. If the errors persist, please share a larger excerpt of your logfile.
```

File: source/_components/plugwise.markdown [hunk 25]
```diff
@@ -0,0 +1,164 @@
...
+      - service: climate.set_temperature
+        data:
+          temperature: 19.5
+```
+
+#### Set preset mode
+
+Service: `climate.set_preset_mode`
+
+Available options include: `home`, `vacation`, `no_frost`, `asleep` & `away`
```
Comment (frenck) [inline source/_components/plugwise.markdown]:
```suggestion
Available options include: `home`, `vacation`, `no_frost`, `asleep` & `away`.
```

**Review by frenck — CHANGES_REQUESTED** [review]
Hi there @CoMPaTech, I've reviewed your PR and left some comments. Could you please take a look? 👍

File: source/_components/plugwise.markdown (hunk 20 above)
Comment (CoMPaTech) [reply source/_components/plugwise.markdown:69]: Is this in-line with what you wanted?

**Comment by CoMPaTech** [comment]
Hi @frenck, thanks for dotting my i's 👍 
Is the full example the correct improvement (left that one unresolved)

File: source/_components/plugwise.markdown (hunk 20 above)
Comment (frenck) [reply source/_components/plugwise.markdown:69]: Yep, that is how it should be done 👍

**Review by frenck — APPROVED** [review]
Thanks for addressing the comments so quickly @CoMPaTech! 👍

**Pull request merged**